packages/server/src/adapters/ws.ts

```diff
@@ -316,10 +333,41 @@ export function getWSConnectionHandler<TRouter extends AnyRouter>(
   };
 }
 
+/**
+ * Handle WebSocket keep-alive messages
+ */
+function handleKeepAlive(
+  client: ws.WebSocket,
+  pingMs = 30000,
+  pongWaitMs = 5000,
+) {
+  let heartbeatTimeout: NodeJS.Timeout | undefined;
+  const heartbeatInterval = setInterval(() => {
+    if (client.readyState !== WEBSOCKET_OPEN) {
+      return;
+    }
+    // First we send a ping message and wait for a pong
+    client.ping();
+    // We set a timeout to close the connection if the pong is not received
+    heartbeatTimeout = setTimeout(() => {
+      client.terminate();
+      clearInterval(heartbeatInterval);
+    }, pongWaitMs);
+  }, pingMs);
```

Comment:
should we use [`unref`](https://nodejs.org/api/timers.html#immediateunref) here?
```suggestion
  }, pingMs).unref()
```